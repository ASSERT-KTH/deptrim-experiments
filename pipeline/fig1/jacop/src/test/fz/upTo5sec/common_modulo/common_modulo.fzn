array [1..7] of int: X_INTRODUCED_37 = [1,-1,-1,-1,-1,-1,-1];
array [1..5] of int: X_INTRODUCED_55 = [1,-1,-1,-1,-1];
var 0..4: a:: output_var:: is_defined_var;
var 0..6: b:: output_var:: is_defined_var;
var set of 0..4: X_INTRODUCED_10 ::var_is_introduced ;
var set of 0..4: X_INTRODUCED_11 ::var_is_introduced ;
var -5..5: X_INTRODUCED_13 ::var_is_introduced ;
var bool: X_INTRODUCED_14 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_15 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_17 ::var_is_introduced ;
var bool: X_INTRODUCED_18 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_19 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_21 ::var_is_introduced ;
var bool: X_INTRODUCED_22 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_23 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_25 ::var_is_introduced ;
var bool: X_INTRODUCED_26 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_27 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_29 ::var_is_introduced ;
var bool: X_INTRODUCED_30 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_31 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_33 ::var_is_introduced ;
var bool: X_INTRODUCED_34 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_35 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_39 ::var_is_introduced ;
var bool: X_INTRODUCED_40 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_41 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_43 ::var_is_introduced ;
var bool: X_INTRODUCED_44 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_45 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_47 ::var_is_introduced ;
var bool: X_INTRODUCED_48 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_49 ::var_is_introduced :: is_defined_var;
var -5..5: X_INTRODUCED_51 ::var_is_introduced ;
var bool: X_INTRODUCED_52 ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_53 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_56 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_57 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_58 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_59 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_60 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_61 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_62 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_63 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_64 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_65 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_66 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_67 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_68 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_69 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_70 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_71 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_72 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_73 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_74 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_75 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_76 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_77 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_78 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_79 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_80 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_81 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_82 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_83 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_84 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_85 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_86 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_87 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_88 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_89 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_90 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_91 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_92 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_93 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_94 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_95 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_96 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_97 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_98 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_99 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_100 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_101 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_102 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_103 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_104 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_105 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_106 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_107 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_108 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_109 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_110 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_111 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_112 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_113 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_114 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_115 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_116 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_117 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_118 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_119 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_120 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_121 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_122 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_123 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_124 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_125 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_126 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_127 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_128 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_129 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_130 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_131 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_132 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_133 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_134 ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_135 ::var_is_introduced :: is_defined_var;
array [1..4] of var int: x:: output_array([1..4]) = [0,4,0,8];
array [1..6] of var int: y:: output_array([1..6]) = [7,5,4,9,2,4];
constraint int_mod(7,5,X_INTRODUCED_13);
constraint int_mod(5,5,X_INTRODUCED_17);
constraint int_mod(4,5,X_INTRODUCED_21);
constraint int_mod(9,5,X_INTRODUCED_25);
constraint int_mod(2,5,X_INTRODUCED_29);
constraint int_mod(4,5,X_INTRODUCED_33);
constraint int_lin_eq(X_INTRODUCED_37,[b,X_INTRODUCED_15,X_INTRODUCED_19,X_INTRODUCED_23,X_INTRODUCED_27,X_INTRODUCED_31,X_INTRODUCED_35],0):: defines_var(b);
constraint int_mod(0,5,X_INTRODUCED_39);
constraint int_mod(4,5,X_INTRODUCED_43);
constraint int_mod(0,5,X_INTRODUCED_47);
constraint int_mod(8,5,X_INTRODUCED_51);
constraint int_lin_eq(X_INTRODUCED_55,[a,X_INTRODUCED_41,X_INTRODUCED_45,X_INTRODUCED_49,X_INTRODUCED_53],0):: defines_var(a);
constraint bool_eq(X_INTRODUCED_62,X_INTRODUCED_64);
constraint bool_eq(X_INTRODUCED_71,X_INTRODUCED_73);
constraint bool_eq(X_INTRODUCED_80,X_INTRODUCED_82);
constraint bool_eq(X_INTRODUCED_89,X_INTRODUCED_91);
constraint bool_eq(X_INTRODUCED_98,X_INTRODUCED_100);
constraint set_in(X_INTRODUCED_13,X_INTRODUCED_11);
constraint set_in(X_INTRODUCED_17,X_INTRODUCED_11);
constraint set_in(X_INTRODUCED_21,X_INTRODUCED_11);
constraint set_in(X_INTRODUCED_25,X_INTRODUCED_11);
constraint set_in(X_INTRODUCED_29,X_INTRODUCED_11);
constraint set_in(X_INTRODUCED_33,X_INTRODUCED_11);
constraint bool_eq(X_INTRODUCED_105,X_INTRODUCED_107);
constraint bool_eq(X_INTRODUCED_112,X_INTRODUCED_114);
constraint bool_eq(X_INTRODUCED_119,X_INTRODUCED_121);
constraint bool_eq(X_INTRODUCED_126,X_INTRODUCED_128);
constraint bool_eq(X_INTRODUCED_133,X_INTRODUCED_135);
constraint set_in(X_INTRODUCED_39,X_INTRODUCED_10);
constraint set_in(X_INTRODUCED_43,X_INTRODUCED_10);
constraint set_in(X_INTRODUCED_47,X_INTRODUCED_10);
constraint set_in(X_INTRODUCED_51,X_INTRODUCED_10);
constraint set_in_reif(X_INTRODUCED_13,X_INTRODUCED_10,X_INTRODUCED_14):: defines_var(X_INTRODUCED_14);
constraint bool2int(X_INTRODUCED_14,X_INTRODUCED_15):: defines_var(X_INTRODUCED_15);
constraint set_in_reif(X_INTRODUCED_17,X_INTRODUCED_10,X_INTRODUCED_18):: defines_var(X_INTRODUCED_18);
constraint bool2int(X_INTRODUCED_18,X_INTRODUCED_19):: defines_var(X_INTRODUCED_19);
constraint set_in_reif(X_INTRODUCED_21,X_INTRODUCED_10,X_INTRODUCED_22):: defines_var(X_INTRODUCED_22);
constraint bool2int(X_INTRODUCED_22,X_INTRODUCED_23):: defines_var(X_INTRODUCED_23);
constraint set_in_reif(X_INTRODUCED_25,X_INTRODUCED_10,X_INTRODUCED_26):: defines_var(X_INTRODUCED_26);
constraint bool2int(X_INTRODUCED_26,X_INTRODUCED_27):: defines_var(X_INTRODUCED_27);
constraint set_in_reif(X_INTRODUCED_29,X_INTRODUCED_10,X_INTRODUCED_30):: defines_var(X_INTRODUCED_30);
constraint bool2int(X_INTRODUCED_30,X_INTRODUCED_31):: defines_var(X_INTRODUCED_31);
constraint set_in_reif(X_INTRODUCED_33,X_INTRODUCED_10,X_INTRODUCED_34):: defines_var(X_INTRODUCED_34);
constraint bool2int(X_INTRODUCED_34,X_INTRODUCED_35):: defines_var(X_INTRODUCED_35);
constraint set_in_reif(X_INTRODUCED_39,X_INTRODUCED_11,X_INTRODUCED_40):: defines_var(X_INTRODUCED_40);
constraint bool2int(X_INTRODUCED_40,X_INTRODUCED_41):: defines_var(X_INTRODUCED_41);
constraint set_in_reif(X_INTRODUCED_43,X_INTRODUCED_11,X_INTRODUCED_44):: defines_var(X_INTRODUCED_44);
constraint bool2int(X_INTRODUCED_44,X_INTRODUCED_45):: defines_var(X_INTRODUCED_45);
constraint set_in_reif(X_INTRODUCED_47,X_INTRODUCED_11,X_INTRODUCED_48):: defines_var(X_INTRODUCED_48);
constraint bool2int(X_INTRODUCED_48,X_INTRODUCED_49):: defines_var(X_INTRODUCED_49);
constraint set_in_reif(X_INTRODUCED_51,X_INTRODUCED_11,X_INTRODUCED_52):: defines_var(X_INTRODUCED_52);
constraint bool2int(X_INTRODUCED_52,X_INTRODUCED_53):: defines_var(X_INTRODUCED_53);
constraint int_ne_reif(X_INTRODUCED_13,0,X_INTRODUCED_56):: defines_var(X_INTRODUCED_56);
constraint int_ne_reif(X_INTRODUCED_17,0,X_INTRODUCED_57):: defines_var(X_INTRODUCED_57);
constraint int_ne_reif(X_INTRODUCED_21,0,X_INTRODUCED_58):: defines_var(X_INTRODUCED_58);
constraint int_ne_reif(X_INTRODUCED_25,0,X_INTRODUCED_59):: defines_var(X_INTRODUCED_59);
constraint int_ne_reif(X_INTRODUCED_29,0,X_INTRODUCED_60):: defines_var(X_INTRODUCED_60);
constraint int_ne_reif(X_INTRODUCED_33,0,X_INTRODUCED_61):: defines_var(X_INTRODUCED_61);
constraint array_bool_and([X_INTRODUCED_56,X_INTRODUCED_57,X_INTRODUCED_58,X_INTRODUCED_59,X_INTRODUCED_60,X_INTRODUCED_61],X_INTRODUCED_62):: defines_var(X_INTRODUCED_62);
constraint set_in_reif(0,X_INTRODUCED_11,X_INTRODUCED_63):: defines_var(X_INTRODUCED_63);
constraint bool_eq_reif(X_INTRODUCED_63,false,X_INTRODUCED_64):: defines_var(X_INTRODUCED_64);
constraint int_ne_reif(X_INTRODUCED_13,1,X_INTRODUCED_65):: defines_var(X_INTRODUCED_65);
constraint int_ne_reif(X_INTRODUCED_17,1,X_INTRODUCED_66):: defines_var(X_INTRODUCED_66);
constraint int_ne_reif(X_INTRODUCED_21,1,X_INTRODUCED_67):: defines_var(X_INTRODUCED_67);
constraint int_ne_reif(X_INTRODUCED_25,1,X_INTRODUCED_68):: defines_var(X_INTRODUCED_68);
constraint int_ne_reif(X_INTRODUCED_29,1,X_INTRODUCED_69):: defines_var(X_INTRODUCED_69);
constraint int_ne_reif(X_INTRODUCED_33,1,X_INTRODUCED_70):: defines_var(X_INTRODUCED_70);
constraint array_bool_and([X_INTRODUCED_65,X_INTRODUCED_66,X_INTRODUCED_67,X_INTRODUCED_68,X_INTRODUCED_69,X_INTRODUCED_70],X_INTRODUCED_71):: defines_var(X_INTRODUCED_71);
constraint set_in_reif(1,X_INTRODUCED_11,X_INTRODUCED_72):: defines_var(X_INTRODUCED_72);
constraint bool_eq_reif(X_INTRODUCED_72,false,X_INTRODUCED_73):: defines_var(X_INTRODUCED_73);
constraint int_ne_reif(X_INTRODUCED_13,2,X_INTRODUCED_74):: defines_var(X_INTRODUCED_74);
constraint int_ne_reif(X_INTRODUCED_17,2,X_INTRODUCED_75):: defines_var(X_INTRODUCED_75);
constraint int_ne_reif(X_INTRODUCED_21,2,X_INTRODUCED_76):: defines_var(X_INTRODUCED_76);
constraint int_ne_reif(X_INTRODUCED_25,2,X_INTRODUCED_77):: defines_var(X_INTRODUCED_77);
constraint int_ne_reif(X_INTRODUCED_29,2,X_INTRODUCED_78):: defines_var(X_INTRODUCED_78);
constraint int_ne_reif(X_INTRODUCED_33,2,X_INTRODUCED_79):: defines_var(X_INTRODUCED_79);
constraint array_bool_and([X_INTRODUCED_74,X_INTRODUCED_75,X_INTRODUCED_76,X_INTRODUCED_77,X_INTRODUCED_78,X_INTRODUCED_79],X_INTRODUCED_80):: defines_var(X_INTRODUCED_80);
constraint set_in_reif(2,X_INTRODUCED_11,X_INTRODUCED_81):: defines_var(X_INTRODUCED_81);
constraint bool_eq_reif(X_INTRODUCED_81,false,X_INTRODUCED_82):: defines_var(X_INTRODUCED_82);
constraint int_ne_reif(X_INTRODUCED_13,3,X_INTRODUCED_83):: defines_var(X_INTRODUCED_83);
constraint int_ne_reif(X_INTRODUCED_17,3,X_INTRODUCED_84):: defines_var(X_INTRODUCED_84);
constraint int_ne_reif(X_INTRODUCED_21,3,X_INTRODUCED_85):: defines_var(X_INTRODUCED_85);
constraint int_ne_reif(X_INTRODUCED_25,3,X_INTRODUCED_86):: defines_var(X_INTRODUCED_86);
constraint int_ne_reif(X_INTRODUCED_29,3,X_INTRODUCED_87):: defines_var(X_INTRODUCED_87);
constraint int_ne_reif(X_INTRODUCED_33,3,X_INTRODUCED_88):: defines_var(X_INTRODUCED_88);
constraint array_bool_and([X_INTRODUCED_83,X_INTRODUCED_84,X_INTRODUCED_85,X_INTRODUCED_86,X_INTRODUCED_87,X_INTRODUCED_88],X_INTRODUCED_89):: defines_var(X_INTRODUCED_89);
constraint set_in_reif(3,X_INTRODUCED_11,X_INTRODUCED_90):: defines_var(X_INTRODUCED_90);
constraint bool_eq_reif(X_INTRODUCED_90,false,X_INTRODUCED_91):: defines_var(X_INTRODUCED_91);
constraint int_ne_reif(X_INTRODUCED_13,4,X_INTRODUCED_92):: defines_var(X_INTRODUCED_92);
constraint int_ne_reif(X_INTRODUCED_17,4,X_INTRODUCED_93):: defines_var(X_INTRODUCED_93);
constraint int_ne_reif(X_INTRODUCED_21,4,X_INTRODUCED_94):: defines_var(X_INTRODUCED_94);
constraint int_ne_reif(X_INTRODUCED_25,4,X_INTRODUCED_95):: defines_var(X_INTRODUCED_95);
constraint int_ne_reif(X_INTRODUCED_29,4,X_INTRODUCED_96):: defines_var(X_INTRODUCED_96);
constraint int_ne_reif(X_INTRODUCED_33,4,X_INTRODUCED_97):: defines_var(X_INTRODUCED_97);
constraint array_bool_and([X_INTRODUCED_92,X_INTRODUCED_93,X_INTRODUCED_94,X_INTRODUCED_95,X_INTRODUCED_96,X_INTRODUCED_97],X_INTRODUCED_98):: defines_var(X_INTRODUCED_98);
constraint set_in_reif(4,X_INTRODUCED_11,X_INTRODUCED_99):: defines_var(X_INTRODUCED_99);
constraint bool_eq_reif(X_INTRODUCED_99,false,X_INTRODUCED_100):: defines_var(X_INTRODUCED_100);
constraint int_ne_reif(X_INTRODUCED_39,0,X_INTRODUCED_101):: defines_var(X_INTRODUCED_101);
constraint int_ne_reif(X_INTRODUCED_43,0,X_INTRODUCED_102):: defines_var(X_INTRODUCED_102);
constraint int_ne_reif(X_INTRODUCED_47,0,X_INTRODUCED_103):: defines_var(X_INTRODUCED_103);
constraint int_ne_reif(X_INTRODUCED_51,0,X_INTRODUCED_104):: defines_var(X_INTRODUCED_104);
constraint array_bool_and([X_INTRODUCED_101,X_INTRODUCED_102,X_INTRODUCED_103,X_INTRODUCED_104],X_INTRODUCED_105):: defines_var(X_INTRODUCED_105);
constraint set_in_reif(0,X_INTRODUCED_10,X_INTRODUCED_106):: defines_var(X_INTRODUCED_106);
constraint bool_eq_reif(X_INTRODUCED_106,false,X_INTRODUCED_107):: defines_var(X_INTRODUCED_107);
constraint int_ne_reif(X_INTRODUCED_39,1,X_INTRODUCED_108):: defines_var(X_INTRODUCED_108);
constraint int_ne_reif(X_INTRODUCED_43,1,X_INTRODUCED_109):: defines_var(X_INTRODUCED_109);
constraint int_ne_reif(X_INTRODUCED_47,1,X_INTRODUCED_110):: defines_var(X_INTRODUCED_110);
constraint int_ne_reif(X_INTRODUCED_51,1,X_INTRODUCED_111):: defines_var(X_INTRODUCED_111);
constraint array_bool_and([X_INTRODUCED_108,X_INTRODUCED_109,X_INTRODUCED_110,X_INTRODUCED_111],X_INTRODUCED_112):: defines_var(X_INTRODUCED_112);
constraint set_in_reif(1,X_INTRODUCED_10,X_INTRODUCED_113):: defines_var(X_INTRODUCED_113);
constraint bool_eq_reif(X_INTRODUCED_113,false,X_INTRODUCED_114):: defines_var(X_INTRODUCED_114);
constraint int_ne_reif(X_INTRODUCED_39,2,X_INTRODUCED_115):: defines_var(X_INTRODUCED_115);
constraint int_ne_reif(X_INTRODUCED_43,2,X_INTRODUCED_116):: defines_var(X_INTRODUCED_116);
constraint int_ne_reif(X_INTRODUCED_47,2,X_INTRODUCED_117):: defines_var(X_INTRODUCED_117);
constraint int_ne_reif(X_INTRODUCED_51,2,X_INTRODUCED_118):: defines_var(X_INTRODUCED_118);
constraint array_bool_and([X_INTRODUCED_115,X_INTRODUCED_116,X_INTRODUCED_117,X_INTRODUCED_118],X_INTRODUCED_119):: defines_var(X_INTRODUCED_119);
constraint set_in_reif(2,X_INTRODUCED_10,X_INTRODUCED_120):: defines_var(X_INTRODUCED_120);
constraint bool_eq_reif(X_INTRODUCED_120,false,X_INTRODUCED_121):: defines_var(X_INTRODUCED_121);
constraint int_ne_reif(X_INTRODUCED_39,3,X_INTRODUCED_122):: defines_var(X_INTRODUCED_122);
constraint int_ne_reif(X_INTRODUCED_43,3,X_INTRODUCED_123):: defines_var(X_INTRODUCED_123);
constraint int_ne_reif(X_INTRODUCED_47,3,X_INTRODUCED_124):: defines_var(X_INTRODUCED_124);
constraint int_ne_reif(X_INTRODUCED_51,3,X_INTRODUCED_125):: defines_var(X_INTRODUCED_125);
constraint array_bool_and([X_INTRODUCED_122,X_INTRODUCED_123,X_INTRODUCED_124,X_INTRODUCED_125],X_INTRODUCED_126):: defines_var(X_INTRODUCED_126);
constraint set_in_reif(3,X_INTRODUCED_10,X_INTRODUCED_127):: defines_var(X_INTRODUCED_127);
constraint bool_eq_reif(X_INTRODUCED_127,false,X_INTRODUCED_128):: defines_var(X_INTRODUCED_128);
constraint int_ne_reif(X_INTRODUCED_39,4,X_INTRODUCED_129):: defines_var(X_INTRODUCED_129);
constraint int_ne_reif(X_INTRODUCED_43,4,X_INTRODUCED_130):: defines_var(X_INTRODUCED_130);
constraint int_ne_reif(X_INTRODUCED_47,4,X_INTRODUCED_131):: defines_var(X_INTRODUCED_131);
constraint int_ne_reif(X_INTRODUCED_51,4,X_INTRODUCED_132):: defines_var(X_INTRODUCED_132);
constraint array_bool_and([X_INTRODUCED_129,X_INTRODUCED_130,X_INTRODUCED_131,X_INTRODUCED_132],X_INTRODUCED_133):: defines_var(X_INTRODUCED_133);
constraint set_in_reif(4,X_INTRODUCED_10,X_INTRODUCED_134):: defines_var(X_INTRODUCED_134);
constraint bool_eq_reif(X_INTRODUCED_134,false,X_INTRODUCED_135):: defines_var(X_INTRODUCED_135);
solve  satisfy;
