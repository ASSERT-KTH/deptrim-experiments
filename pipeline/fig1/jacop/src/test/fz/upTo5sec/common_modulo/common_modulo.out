a = 3;
b = 4;
x = array1d(1..4, [0, 4, 0, 8]);
y = array1d(1..6, [7, 5, 4, 9, 2, 4]);
----------